modules/Core/pages/panel/users_sessions.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/*
+ *  Made by Supercrafter100
+ *  https://github.com/NamelessMC/Nameless/
+ *  NamelessMC version 2.0.3
+ *
+ *  License: MIT
+ *
+ *  Panel user sessions page
+ */
+if (!$user->handlePanelPageLoad('admincp.users.sessions')) {
+    require_once(ROOT_PATH . '/403.php');
+    die();
+}
+
+const PAGE = 'panel';
+const PARENT_PAGE = 'users';
+const PANEL_PAGE = 'sessions';
+$page_title = $language->get('admin', 'sessions');
+require_once(ROOT_PATH . '/core/templates/backend_init.php');
+
+// Load modules + template
+Module::loadPage($user, $pages, $cache, $smarty, [$navigation, $cc_nav, $staffcp_nav], $widgets, $template);
+
+if (!isset($_GET['id']) || !is_numeric($_GET['id'])) {
+    Redirect::to(URL::build('/panel/users'));
+}
+
+$view_user = new User($_GET['id']);
+if (!$view_user->exists()) {
+    Redirect::to(URL::build('/panel/users'));
+}
+
+if (Input::exists()) {
+    if (Token::check()) {
+        if ($_POST['action'] == 'logout' && isset($_POST['sid'])) {
+            $id = $_POST['sid'];
+            DB::getInstance()->update('users_session', ['id', $id], [
+                'active' => false
+            ]);
+            $success = $language->get('general', 'logout_session_successfully');
+        }
+    } else {
+        $errors[] = $language->get('general', 'invalid_token');
+    }
+}
+$timeago = new TimeAgo(TIMEZONE);
+$sessions = DB::getInstance()->query('SELECT * FROM nl2_users_session WHERE user_id = ?', [$view_user->data()->id])->results();
+$user_sessions_list = [];
+
+$cache->setCache('ip_location_results');
+foreach ($sessions as $session) {
+    if ($session->ip && $cache->retrieve($session->ip)) {
+        $location = $cache->retrieve($session->ip);
+    } else if ($session->ip) {
+        $response = HttpClient::get('http://ip-api.com/json/' . $session->ip);
```

Comment:
Can this be `https` ?

Reply:
Also I'm wondering if there are any privacy requirements around sharing IP addresses with third parties?

Reply:
I think this would be fine - we are not sending any other information *about* the IP to this service (username, email, etc), just the IP itself.
By visiting a page users are implicitly consenting to sharing their IP 🤷🏼

Reply:
> Can this be `https` ?

Heh, unfortunately not, only `http` is allowed on the free/unregistered plan of `ip-api.com`. I think this is partially why Derk is thinking about a country db

Reply:
I https worked on my site when i added this to MCStatistics and i believe we still need to add to tos or policy that we share the ip to that site

Reply:
The one @Derkades linked looks great. Less than 10MB, whereas GeoLite2 is closer to 70MB.